torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -304,20 +227,38 @@ static THPObjectPtr wrapTuple(PyObject* index) {
   return res;
 }
 
+// NOTE: You might ask "why can't we convert all Python indices into an `std::vector<TensorIndex>`,
+// and then simply call the C++ multi-dim indexing function?" The answer is that the conversion
+// in general adds an overhead of 100s of nanoseconds to indexing calls such as `tensor[0]`
+// and `tensor[0, 0]`, which attributes to 10%-30% increase in runtime and thus undesirable.
```

Comment:
Umm, actually, no, I would not ask that question reading this code XD Where would I allocate a vector? It doesn't seem like there is any logical point where this would happen; not here, anyway.